collect_app/src/main/java/org/odk/collect/android/widgets/BaseStringWidget.java

```diff
@@ -63,8 +65,13 @@ public void clearAnswer() {
 
     @Override
     public IAnswerData getAnswer() {
-        String s = getAnswerText();
-        return !s.equals("") ? new StringData(s) : null;
+        if (this instanceof IntegerWidget || this instanceof ExIntegerWidget) {
```

Comment:
Same comment as before -- these conditionals indicate you need another level in your hierarchy.